### DISCUSSION TITLE
Need help with menu bar for Tiptap v2

### DISCUSSION THREAD

**Discussion by mpemburn** [ueberdosis/tiptap#2384] [General Questions] [answered]
Hi,

We've been using Tiptap v1 for some time, and we're on the verge of upgrading to Vue 3 and Tiptap v2.  I've started a test Laravel project and installed v2 with the Starter Kit.  In order to retain the appearance and functionality of our current Tiptap editors, I need to make a menu bar with buttons.  I can see how to include a button, and make it work.  For example, this works just fine:
```
<button @click="editor.chain().focus().toggleBold().run()">
  Bold
</button>
```
However, I need to be able to indicate when the style is active so, based on the docs, I've change it to:
```
<button :class="{ 'is-active': editor.isActive('bold') }" @click="editor.chain().focus().toggleBold().run()">
    Bold
</button>
```
...which bombs with the error:

`Uncaught TypeError: Cannot read properties of null (reading 'isActive')`

Can someone tell me what I need to do to make this work?

Thanks in advance,

Mark

**Comment by philippkuehn** [comment, marked as answer]
The editor is not available on the first render. This helps:

```vue
<template v-if="editor">
  <button :class="{ 'is-active': editor.isActive('bold') }" @click="editor.chain().focus().toggleBold().run()">
      Bold
  </button>
  <!-- ... -->
</template>
```

**Reply by mpemburn** [reply]
Thanks for your quick reply!  I tried the change, and it still gets that same error.  This is the whole component:
```
<template v-if="editor">
    <button :class="{ 'is-active': editor.isActive('bold') }" @click="editor.chain().focus().toggleBold().run()">
        Bold
    </button>
    <editor-content :editor="editor"/>
</template>

<script>
import {Editor, EditorContent} from '@tiptap/vue-3'
import StarterKit from '@tiptap/starter-kit'

export default {
    components: {
        EditorContent,
    },

    data() {
        return {
            editor: null,
            html: null
        }
    },
    mounted() {
        this.editor = new Editor({
            content: this.html,
            extensions: [
                StarterKit,
            ],
        })
    },

    beforeUnmount() {
        this.editor.destroy()
    },
}
</script>
```

**Reply by philippkuehn** [reply]
the root `<template>` in your SFC can’t get any attribute.

```vue
<template>
  <template v-if="editor">
    <button :class="{ 'is-active': editor.isActive('bold') }" @click="editor.chain().focus().toggleBold().run()">
      Bold
    </button>
  </template>
  <editor-content :editor="editor"/>
</template>

<script>
// ...
</script>
```

**Comment by mpemburn** [comment]
Aha!  That did the trick—thanks!  I still have a lot to learn about Vue...
